File: prisma/seed.ts
import { PrismaClient } from "@prisma/client";
import bcrypt from "bcryptjs";

const prisma = new PrismaClient();

async function main() {
  const now = new Date().toISOString();

  const adminPassword = await bcrypt.hash("Admin123", 12);
  const demoPassword = await bcrypt.hash("Demo123", 12);

  const admin = await prisma.user.upsert({
    where: { email: "admin@plancraft.ai" },
    update: {},
    create: {
      name: "Admin",
      email: "admin@plancraft.ai",
      password: adminPassword,
      role: "admin",
      plan: "enterprise",
      verified: true,
      company: "PlanCraftAI",
      country: "US",
      aiCreditsUsed: 0,
      aiCreditsTotal: 9999,
      storageUsedMb: 0,
      storageQuotaMb: 102400,
      projectsCount: 0,
      createdAt: now,
      updatedAt: now,
    },
  });

  const demo = await prisma.user.upsert({
    where: { email: "demo@plancraft.ai" },
    update: {},
    create: {
      name: "Demo User",
      email: "demo@plancraft.ai",
      password: demoPassword,
      role: "user",
      plan: "pro",
      verified: true,
      company: "Demo Corp",
      country: "US",
      aiCreditsUsed: 5,
      aiCreditsTotal: 100,
      storageUsedMb: 120,
      storageQuotaMb: 10240,
      projectsCount: 0,
      createdAt: now,
      updatedAt: now,
    },
  });

  console.log("Seeded admin:", admin.email);
  console.log("Seeded demo:", demo.email);

  const sampleProject = await prisma.project.upsert({
    where: { id: "seed-p1" },
    update: {},
    create: {
      id: "seed-p1",
      name: "Modern Luxury Villa",
      description: "A 5-bedroom modern villa with open plan living",
      userId: demo.id,
      plotLength: 60,
      plotWidth: 40,
      facing: "East",
      floors: 2,
      budgetTier: "Premium",
      style: "Modern",
      vastu: true,
      status: "completed",
      shared: true,
      viewCount: 1243,
      vastuScore: 92,
      sustainabilityScore: 78,
      stars: 4,
      createdAt: now,
      updatedAt: now,
    },
  });

  await prisma.room.createMany({
    data: [
      { name: "Living Room", width: 20, length: 25, level: 0, type: "living", area: 500, projectId: sampleProject.id },
      { name: "Master Bedroom", width: 16, length: 18, level: 0, type: "bedroom", area: 288, projectId: sampleProject.id },
      { name: "Kitchen", width: 12, length: 15, level: 0, type: "kitchen", area: 180, projectId: sampleProject.id },
      { name: "Bedroom 2", width: 14, length: 14, level: 1, type: "bedroom", area: 196, projectId: sampleProject.id },
      { name: "Bedroom 3", width: 13, length: 14, level: 1, type: "bedroom", area: 182, projectId: sampleProject.id },
    ],
  });

  await prisma.costEstimate.upsert({
    where: { projectId: sampleProject.id },
    update: {},
    create: {
      foundation: 45000,
      concrete: 32000,
      steel: 28000,
      brick: 18000,
      flooring: 35000,
      plumbing: 15000,
      electrical: 12000,
      labor: 40000,
      contingency: 10000,
      designFees: 8000,
      total: 243000,
      projectId: sampleProject.id,
    },
  });

  console.log("Seeded sample project:", sampleProject.name);
}

main()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

File: backend/app/seed.py
"""
Database Seeder - Creates initial data for development.
"""
import asyncio
from app.database import async_session_factory
from app.utils.security import hash_password
from app.models.user import User
from app.models.subscription import SubscriptionPlan
from app.models.settings import Settings
from app.models.template import Template
from app.models.project import Project
from app.models.room import Room
from app.models.notification import Notification
from sqlalchemy import select
import uuid
import logging

logger = logging.getLogger(__name__)


async def seed():
    async with async_session_factory() as db:
        # Check if already seeded
        result = await db.execute(select(User).limit(1))
        if result.scalar_one_or_none():
            logger.info("Database already seeded, skipping.")
            return
        
        # Create admin user
        admin = User(
            name="Admin",
            email="admin@plancraft.ai",
            password=hash_password("Admin123"),
            role="super_admin",
            plan="enterprise",
            verified=True,
            is_active=True,
            ai_credits_total=999999,
            storage_quota_mb=999999,
        )
        db.add(admin)
        
        # Create demo user
        demo = User(
            name="Demo User",
            email="demo@plancraft.ai",
            password=hash_password("Demo123"),
            role="user",
            plan="pro",
            verified=True,
            is_active=True,
            ai_credits_total=100,
            storage_quota_mb=10240,
        )
        db.add(demo)
        
        # Create subscription plans
        plans = [
            SubscriptionPlan(name="free", price_monthly=0, price_yearly=0, credits=5, storage_mb=500, max_projects=3,
                             features={"ai_generation": True, "basic_exports": True, "community_access": True}),
            SubscriptionPlan(name="pro", price_monthly=2499, price_yearly=24990, credits=100, storage_mb=10240, max_projects=50,
                             features={"ai_generation": True, "hd_exports": True, "vastu_analysis": True, "priority_support": True}),
            SubscriptionPlan(name="business", price_monthly=7999, price_yearly=79990, credits=500, storage_mb=51200, max_projects=200,
                             features={"ai_generation": True, "team_collaboration": True, "api_access": True, "custom_templates": True}),
            SubscriptionPlan(name="enterprise", price_monthly=19999, price_yearly=199990, credits=2000, storage_mb=512000, max_projects=9999,
                             features={"all_features": True, "dedicated_support": True, "white_label": True, "custom_integrations": True}),
        ]
        for plan in plans:
            db.add(plan)
        
        # Create demo project
        project = Project(
            name="My Dream Home",
            description="A modern 3BHK home with Vastu compliance",
            user_id=demo.id,
            plot_length=60,
            plot_width=40,
            facing="East",
            floors=2,
            budget_tier="Premium",
            style="Modern",
            vastu=True,
            status="completed",
        )
        db.add(project)
        
        # Create rooms for demo project
        rooms_data = [
            ("Master Bedroom", 14, 16, 0, "bedroom", "#e0f2fe"),
            ("Bedroom 2", 12, 14, 0, "bedroom", "#faf5ff"),
            ("Bedroom 3", 12, 14, 1, "bedroom", "#faf5ff"),
            ("Living Room", 16, 22, 0, "living", "#f0fdf4"),
            ("Dining Room", 12, 14, 0, "dining", "#fef3c7"),
            ("Kitchen", 10, 12, 0, "kitchen", "#fee2e2"),
            ("Bathroom 1", 6, 8, 0, "bathroom", "#f1f5f9"),
            ("Bathroom 2", 6, 8, 1, "bathroom", "#f1f5f9"),
            ("Pooja Room", 6, 6, 0, "pooja", "#fef3c7"),
        ]
        for name, w, l, level, rtype, color in rooms_data:
            room = Room(project_id=project.id, name=name, width=w, length=l,
                        level=level, room_type=rtype, color=color)
            db.add(room)
        
        # Create templates
        templates = [
            Template(name="Modern 3BHK", description="Contemporary 3-bedroom home with open floor plan",
                     category="residential", is_premium=False,
                     data={"bedrooms": 3, "style": "Modern", "area": 1800}),
            Template(name="Premium Villa", description="Luxury villa with pool and garden",
                     category="residential", is_premium=True, price=2499,
                     data={"bedrooms": 4, "style": "Contemporary", "area": 3500}),
            Template(name="Small Office", description="Efficient office space for small teams",
                     category="commercial", is_premium=False,
                     data={"type": "office", "area": 1200}),
        ]
        for t in templates:
            db.add(t)
        
        await db.flush()
        logger.info("Database seeded successfully!")


if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO)
    asyncio.run(seed())
